src/cmd/resources/add_files.rb

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/ruby
+
+# Copyright (C) 2022 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'erb'
+require 'open-uri'
+require 'yaml'
+require 'ostruct'
+
+class Add_files
+  
+  def cmake_lists(resources_dir,open_struct_opt,options)
+    cmakelists_render = ERB.new(File.read(resources_dir+"/CMakeLists.erb"),trim_mode: ">")
+    cmakelists_file = File.new(options['name']+"/CMakeLists.txt", "w")
+    cmakelists_file.puts(cmakelists_render.result(open_struct_opt))
+    cmakelists_file.close
+  end  
+
+  def package_xml(resources_dir,open_struct_opt,options)
+    package_xml_render = ERB.new(File.read(resources_dir+"/package_xml.erb"),trim_mode: ">")
```

Comment:
Any place where you are adding a string in code is considered hard-coding and should only be used as alast resort. I recommend specifying these files in the config itself.